MobileCenterAnalytics/MobileCenterAnalytics/MSAnalytics.m

```diff
@@ -154,64 +154,74 @@ + (BOOL)isAutoPageTrackingEnabled {
 
 #pragma mark - Private methods
 
-- (BOOL)validateEventName:(NSString *)eventName {
+- (BOOL)validateEventName:(NSString *)eventName forLogType:(NSString *)logType {
   if (!eventName || [eventName length] < minEventNameLength) {
     MSLogError([MSAnalytics logTag],
-               @"Event name cannot be null or empty");
+               @"%@ name cannot be null or empty", logType);
     return NO;
   }
   if ([eventName length] > maxEventNameLength) {
     MSLogError([MSAnalytics logTag],
-               @"Event name length cannot be longer then %d characters", minEventNameLength);
+               @"%@ '%@' : name length cannot be longer than %d characters", logType, eventName, maxEventNameLength);
     return NO;
   }
   return YES;
 }
 
-- (NSDictionary<NSString *, NSString *> *)validateProperties:(NSDictionary<NSString *, NSString *> *)properties {
+- (NSDictionary<NSString *, NSString *> *)validateProperties:(NSDictionary<NSString *, NSString *> *)properties
+                                                  forLogName:(NSString *)logName
+                                                     andType:(NSString *)logType {
   NSMutableDictionary<NSString *, NSString *> *validProperties = [NSMutableDictionary new];
   for (id key in properties) {
+
+    // Don't send more properties than we can
```

Comment:
Add a `.` at the end. Sorry to bother, it's just our convention but once used to it it's automatic ;).